metrics.go

```diff
@@ -0,0 +1,207 @@
+package metrics
+
+import (
+	"encoding/json"
+	"os"
+
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/alexcesaro/statsd"
+	"github.com/transcovo/go-chpr-logger"
+)
+
+var defaultClientBuilder *clientBuilder
+
+func defaultClientErrorHandler(err error) {
+	// Do we choose to make the metrics depend on the logger ?
+	// That sounds like a bad idea to me but this would be useful to have logs about the connection to the statsd server
+	// fmt.Println("Error caught in metrics", error)
+	logger.WithField("error", err).Error("[Metrics] Error caught in metrics")
+}
+
+func init() {
+	defaultClientBuilder = &clientBuilder{
+		errorLogger: defaultClientErrorHandler,
+	}
+}
+
+/*
+Struct responsible for keeping all the statsd clients
+*/
+type metricsSender struct {
+	Clients []*statsd.Client
+}
+
+var metrics *metricsSender = &metricsSender{}
```

Comment:
2 - removed
1 - agreed, I'm gonna do that. More generally, I reckon this is a common flaw in our go code, hard to track all structs, methods, interface